rmf_site_editor/src/interaction/preview.rs

```diff
@@ -0,0 +1,166 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+use bevy::prelude::*;
+use bevy::render::camera::{Projection, RenderTarget};
+use bevy::utils::HashMap;
+use bevy::window::{CreateWindow, PresentMode, WindowClosed, WindowId, Windows};
+
+use rmf_site_format::{NameInSite, PhysicalCameraProperties, PreviewableMarker};
+
+/// Instruction to spawn a preview for the given entity
+/// TODO None to encode "Clear all"
+#[derive(Debug, Clone, Copy, Default, PartialEq)]
+pub struct SpawnPreview {
+    pub entity: Option<Entity>,
+}
+
+impl SpawnPreview {
+    pub fn new(entity: Option<Entity>) -> Self {
+        Self { entity }
+    }
+}
+
+/// Used to keep track of what Camera is being previewed in what window for runtime updates
+#[derive(Default)]
+pub struct CameraPreviewWindows(HashMap<Entity, WindowId>);
+
+fn create_camera_window(
+    commands: &mut Commands,
+    entity: Entity,
+    camera_name: &String,
+    camera_properties: &PhysicalCameraProperties,
+    create_window_events: &mut EventWriter<CreateWindow>,
+) -> WindowId {
+    let window_id = WindowId::new();
+    create_window_events.send(CreateWindow {
+        id: window_id,
+        descriptor: WindowDescriptor {
+            width: camera_properties.width as f32,
+            height: camera_properties.height as f32,
+            present_mode: PresentMode::AutoNoVsync,
+            title: "Camera preview: ".to_string() + camera_name,
+            ..default()
+        },
+    });
+    // Now spawn the camera
+    commands.entity(entity).insert(Camera {
+        target: RenderTarget::Window(window_id),
+        is_active: true,
+        ..default()
+    });
+    window_id
+}
+
+// TODO consider renaming this manage_camera_previews and
+// use other systems for other previews
+pub fn manage_previews(
+    mut commands: Commands,
+    mut preview_events: EventReader<SpawnPreview>,
+    previewable: Query<
+        (
+            Entity,
+            &Children,
+            &NameInSite,
+            Option<&PhysicalCameraProperties>,
+        ),
+        With<PreviewableMarker>,
+    >,
+    mut camera_children: Query<(Entity, &mut Projection), With<Camera>>,
+    mut create_window_events: EventWriter<CreateWindow>,
+    mut preview_windows: ResMut<CameraPreviewWindows>,
+) {
+    for event in preview_events.iter() {
+        if let None = event.entity {
+            // TODO clear all previews
+        }
+        if let Some(e) = event.entity {
+            if let Some(open_window) = preview_windows.0.get_mut(&e) {
```

Comment:
If `CameraPreviewWindow` were a component then you could add an `Option<&CameraPreviewWindow>` to the `previewable` query and check its value after querying.